File: package/main/src/Date/dateRange.ts
import { OneDayMs } from "@/Consts/clock";

/**
 * Generate an array containing all dates between the specified start and end dates.
 *
 * @param {Date} startDate - The start date of the range
 * @param {Date} endDate - The end date of the range
 * @returns {Date[]} An array of Date objects from startDate to endDate (inclusive)
 * @example
 * dateRange(new Date('2025-01-01'), new Date('2025-01-03'))
 * // Returns array of dates: [2025-01-01, 2025-01-02, 2025-01-03]
 */
export const dateRange = (startDate: Date, endDate: Date): Date[] => {
  const dates: Date[] = [];
  let currentDate = startDate;
  while (currentDate <= endDate) {
    dates.push(new Date(currentDate));
    currentDate = new Date(currentDate.getTime() + OneDayMs);
  }
  return dates;
};

File: package/umt_python/src/date/date_range.py
from datetime import datetime, timedelta


def date_range(start_date: datetime, end_date: datetime) -> list[datetime]:
    """
    Generate a list containing all dates between the specified start and end dates.

    Args:
        start_date: The start date of the range
        end_date: The end date of the range

    Returns:
        A list of datetime objects from start_date to end_date (inclusive)

    Example:
        >>> from datetime import datetime
        >>> date_range(datetime(2025, 1, 1), datetime(2025, 1, 3))
        [datetime(2025, 1, 1), datetime(2025, 1, 2), datetime(2025, 1, 3)]
    """
    dates: list[datetime] = []
    current_date = start_date
    while current_date <= end_date:
        dates.append(current_date)
        current_date = current_date + timedelta(days=1)
    return dates
